photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/OWLQNTest.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.optimization
+
+import java.util.Random
+
+import breeze.linalg.DenseVector
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.test.Assertions
+import org.testng.Assert
+import org.testng.annotations.{DataProvider, Test}
+
+/**
+  * Test that OWLQN can shrink the coefficients to zero.
+  *
+  * The objective function is (x-1)^2 + (y-4)^2, with L1 regularization C(|x| + |y|). We are only interested in the
+  * behavior around the minimum, where x > 0 and y > 0. Thus the function to be optimized then become
+  * (x-1)^2 + (y-4)^2 + C(x+y) which has obvious analytic solution. The test is based on this function and verifies the
+  * shrinkage of x and y.
+  */
+class OWLQNTest {
```

Comment:
@joshvfleming 
More-or-less a copy of LBFGSTest with the other half of the tests.